wahanegi/vibereport
# CI-551: [Snyk] Fix for 4 vulnerabilities

[![CI-551](https://badgen.net/badge/JIRA/CI-551/0052CC)](https://cloverpop-internship.atlassian.net/browse/CI-551) [<img width="16" alt="Powered by Pull Request Badge" src="https://user-images.githubusercontent.com/1393946/111216524-d2bb8e00-85d4-11eb-821b-ed4c00989c02.png">](https://pullrequestbadge.com/?utm_medium=github&utm_source=wahanegi&utm_campaign=badge_info)

<p>This PR was automatically created by Snyk using the credentials of a real user.</p><br /><h3>Snyk has created this PR to fix one or more vulnerable packages in the `rubygems` dependencies of this project.</h3>



#### Changes included in this PR

- Changes to the following files to upgrade the vulnerable dependencies to a fixed version:
    - Gemfile
    - Gemfile.lock



#### Vulnerabilities that will be fixed
##### With an upgrade:
Severity                   | Priority Score (*)                   | Issue                   | Breaking Change                   | Exploit Maturity
:-------------------------:|-------------------------|:-------------------------|:-------------------------|:-------------------------
![medium severity](https://res.cloudinary.com/snyk/image/upload/w_20,h_20/v1561977819/icon/m.png "medium severity")  |  **449/1000**  <br/> **Why?** Has a fix available, CVSS 4.7  | Cross-site Scripting (XSS) <br/>[SNYK-RUBY-ACTIONPACK-5741907](https://snyk.io/vuln/SNYK-RUBY-ACTIONPACK-5741907) |  No  | No Known Exploit 
![medium severity](https://res.cloudinary.com/snyk/image/upload/w_20,h_20/v1561977819/icon/m.png "medium severity")  |  **429/1000**  <br/> **Why?** Has a fix available, CVSS 4.3  | Information Exposure <br/>[SNYK-RUBY-ACTIVESUPPORT-5851458](https://snyk.io/vuln/SNYK-RUBY-ACTIVESUPPORT-5851458) |  No  | No Known Exploit 
![medium severity](https://res.cloudinary.com/snyk/image/upload/w_20,h_20/v1561977819/icon/m.png "medium severity")  |  **616/1000**  <br/> **Why?** Proof of Concept exploit, Has a fix available, CVSS 5.9  | Web Cache Poisoning <br/>[SNYK-RUBY-RACK-1061917](https://snyk.io/vuln/SNYK-RUBY-RACK-1061917) |  No  | Proof of Concept 
![medium severity](https://res.cloudinary.com/snyk/image/upload/w_20,h_20/v1561977819/icon/m.png "medium severity")  |  **429/1000**  <br/> **Why?** Has a fix available, CVSS 4.3  | Information Exposure <br/>[SNYK-RUBY-RAILTIES-5851410](https://snyk.io/vuln/SNYK-RUBY-RAILTIES-5851410) |  No  | No Known Exploit 

(*) Note that the real score may have changed since the PR was raised.











Check the changes in this PR to ensure they won't cause issues with your project.



------------



**Note:** *You are seeing this because you or someone else with access to this repository has authorized Snyk to open fix PRs.*

For more information:  <img src="https://api.segment.io/v1/pixel/track?data=eyJ3cml0ZUtleSI6InJyWmxZcEdHY2RyTHZsb0lYd0dUcVg4WkFRTnNCOUEwIiwiYW5vbnltb3VzSWQiOiI1OTBjNGFhMS0zYjU2LTQwZDUtODg2ZS0zOGYzNDZkOGE1YjYiLCJldmVudCI6IlBSIHZpZXdlZCIsInByb3BlcnRpZXMiOnsicHJJZCI6IjU5MGM0YWExLTNiNTYtNDBkNS04ODZlLTM4ZjM0NmQ4YTViNiJ9fQ==" width="0" height="0"/>
🧐 [View latest project report](https://app.snyk.io/org/rogergraves/project/6998e670-d362-4fcf-a4ad-ddff8c06149d?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr)

🛠 [Adjust project settings](https://app.snyk.io/org/rogergraves/project/6998e670-d362-4fcf-a4ad-ddff8c06149d?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr/settings)

📚 [Read more about Snyk's upgrade and patch logic](https://support.snyk.io/hc/en-us/articles/360003891078-Snyk-patches-to-fix-vulnerabilities)

[//]: # (snyk:metadata:{"prId":"590c4aa1-3b56-40d5-886e-38f346d8a5b6","prPublicId":"590c4aa1-3b56-40d5-886e-38f346d8a5b6","dependencies":[{"name":"devise","from":"4.9.2","to":"4.9.3"},{"name":"stimulus-rails","from":"1.2.2","to":"1.3.0"},{"name":"turbo-rails","from":"1.4.0","to":"1.5.0"}],"packageManager":"rubygems","projectPublicId":"6998e670-d362-4fcf-a4ad-ddff8c06149d","projectUrl":"https://app.snyk.io/org/rogergraves/project/6998e670-d362-4fcf-a4ad-ddff8c06149d?utm_source=github&utm_medium=referral&page=fix-pr","type":"auto","patch":[],"vulns":["SNYK-RUBY-ACTIONPACK-5741907","SNYK-RUBY-ACTIVESUPPORT-5851458","SNYK-RUBY-RACK-1061917","SNYK-RUBY-RAILTIES-5851410"],"upgrade":["SNYK-RUBY-ACTIONPACK-5741907","SNYK-RUBY-ACTIVESUPPORT-5851458","SNYK-RUBY-RACK-1061917","SNYK-RUBY-RAILTIES-5851410"],"isBreakingChange":false,"env":"prod","prType":"fix","templateVariants":["priorityScore"],"priorityScoreList":[449,429,616,429],"remediationStrategy":"vuln"})

---
## How to test
Please run:
- rake notifications:emotion_selection_invite_all
- rake notifications:remind_checkin
- rake notifications:send_results_email
After please test the login process through the magic link. Ensure that the links are generated correctly and that users can log in to the system through them without any problems. 
---

**Learn how to fix vulnerabilities with free interactive lessons:**

 🦉 [Cross-site Scripting (XSS)](https://learn.snyk.io/lesson/dom-based-xss/?loc&#x3D;fix-pr)

[CI-551]: https://cloverpop-internship.atlassian.net/browse/CI-551?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
